opslevel/resource_opslevel_team_membership.go

```diff
@@ -0,0 +1,180 @@
+	teamId := d.Get("team_id").(string)
+	members := expandStringArray(d.Get("members").(*schema.Set).List())
+
+	team, err := getTeamFromId(teamId, client)
+	if err != nil {
+		return err
+	}
+
+	if err = addMembers(team, members, client); err != nil {
+		return err
+	}
+
+	d.SetId(resource.PrefixedUniqueId(""))
+
+	return resourceTeamMembershipRead(d, client)
+}
+
+func resourceTeamMembershipRead(d *schema.ResourceData, client *opslevel.Client) error {
+	teamId := d.Get("team_id").(string)
+	members, err := getCurrentMembersFromTeamId(teamId, client)
+	if err != nil {
+		return nil
+	}
+
+	d.Set("members", members)
+
+	return nil
+}
+
+func resourceTeamMembershipUpdate(d *schema.ResourceData, client *opslevel.Client) error {
+	if d.HasChange("members") {
+		teamId := d.Get("team_id").(string)
+		o, n := d.GetChange("members")
+		if o == nil {
+			o = new(schema.Set)
+		}
+		if n == nil {
+			n = new(schema.Set)
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		removeSlice := expandStringArray(os.Difference(ns).List())
+		addSlice := expandStringArray(ns.Difference(os).List())
+
+		team, err := getTeamFromId(teamId, client)
+		if err != nil {
+			return err
+		}
+
+		if len(removeSlice) != 0 {
+			if err = removeMembers(team, removeSlice, client); err != nil {
+				return err
+			}
+		}
+
+		if len(addSlice) != 0 {
+			if err = addMembers(team, addSlice, client); err != nil {
+				return err
+			}
+		}
+	}
+	return resourceTeamMembershipRead(d, client)
+}
+
+func resourceTeamMembershipDelete(d *schema.ResourceData, client *opslevel.Client) error {
+	teamId := d.Get("team_id").(string)
+	members := expandStringArray(d.Get("members").(*schema.Set).List())
+
+	team, err := getTeamFromId(teamId, client)
+	if err != nil {
+		return err
+	}
+
+	if err = removeMembers(team, members, client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceTeamMembershipImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*opslevel.Client)
+	teamId := d.Id()
+	currentMembers, err := getCurrentMembersFromTeamId(teamId, client)
+	if err != nil {
+		return nil, err
+	}
+
+	d.Set("team_id", teamId)
+	d.Set("members", currentMembers)
+
+	d.SetId(resource.PrefixedUniqueId(""))
+	return []*schema.ResourceData{d}, nil
+}
+
+func getTeamFromId(teamId string, client *opslevel.Client) (*opslevel.Team, error) {
+	team, err := client.GetTeam(teamId)
+	if err != nil {
+		return nil, err
+	}
+	return team, nil
+}
+
+func getCurrentMembersFromTeamId(teamId string, client *opslevel.Client) ([]string, error) {
+	team, err := getTeamFromId(teamId, client)
+	if err != nil {
+		return nil, err
+	}
+
+	members, err := collectMembersForTeam(team, client)
+	if err != nil {
+		return nil, err
+	}
+	return members, nil
+}
+
+func collectMembersForTeam(team *opslevel.Team, client *opslevel.Client) ([]string, error) {
+	if err := team.Members.Hydrate(team.Id, client); err != nil {
```

Comment:
This should have been automatically called when using `client.GetTeam` - If that was not the case then we have a bug in opslevel-go.

This warrants deeper investigation because we should not need to call this here.

Reply:
Confirmed that calling `GetTeamWithAlias` contains the member nodes.

Reply:
Awesome - I was pretty sure we made opslevel-go handle all the pagination and requesting of sub resources.  FWIW - we are starting to walk this back as we've found it makes a significant impact on our API and it would have been better to code it up as JIT API calls IE:

```
team, err := client.GetTeamWithAlias("MyTeam")

members, err := team.Members(client)
```

But that's a future problem.